src/main/common/parse.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software. You can redistribute this software
+ * and/or modify this software under the terms of the GNU General
+ * Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "parse.h"
+
+#include <errno.h>
+#include <stdlib.h>
+
+#include "platform.h"
+
+static inline intParseResult_t intParseResultOk(long int val, const char *next)
+{
+    return (intParseResult_t) {
+        .status = INT_PARSE_STATUS_OK,
+        .value = val,
+        .next = next,
+    };
+}
+
+static inline intParseResult_t intParseResultErr(enum intParseStatus_e err)
```

Comment:
One thing I don't like is that now it is possible to pass `INT_PARSE_STATUS_OK` here

Reply:
If you really insist, you can use `err ? err : INT_PARSE_ERROR_INTERNAL_ERROR`. But it takes considerable effort(or ignorance) to exploit this and there are tons of other possibilities to do something stupid in BF (an in C generally)